From ab67b39257be83eaab87c4a938c4c58b5cccad7e Mon Sep 17 00:00:00 2001
From: Jimi Huotari <chiitoo@gentoo.org>
Date: Fri, 21 Jun 2019 17:48:32 +0300
Subject: [PATCH] cmake: Install 'libobs.pc' under the correct 'libdir'

In 'libobs/CMakeLists.txt', use '${CMAKE_INSTALL_LIBDIR}' instead of
'${CMAKE_INSTALL_PREFIX}/lib', as the latter results into 'libobs.pc'
being installed under '/lib' when '/lib64' would be more appropriate.

In 'libobs/libobs.pc.in', use '@CMAKE_INSTALL_FULL_LIBDIR@' for
'libdir', '@CMAKE_INSTALL_FULL_INCLUDEDIR@' for 'includedir',
and '@CMAKE_INSTALL_PREFIX@' for 'prefix'.

Gentoo-Bug: https://bugs.gentoo.org/644538
---
 libobs/CMakeLists.txt | 2 +-
 libobs/libobs.pc.in   | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/libobs/CMakeLists.txt b/libobs/CMakeLists.txt
index fed3e694..86811481 100644
--- a/libobs/CMakeLists.txt
+++ b/libobs/CMakeLists.txt
@@ -454,7 +454,7 @@ if(UNIX AND NOT APPLE)
 		set(PRIVATE_LIBS "${PRIVATE_LIBS} -l${LIB}")
 	endforeach()
 	CONFIGURE_FILE("libobs.pc.in" "libobs.pc" @ONLY)
-	install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libobs.pc" DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+	install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libobs.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 endif()
 
 set_target_properties(libobs PROPERTIES
diff --git a/libobs/libobs.pc.in b/libobs/libobs.pc.in
index 03fe4cdd..f6c471ba 100644
--- a/libobs/libobs.pc.in
+++ b/libobs/libobs.pc.in
@@ -1,7 +1,7 @@
-prefix=@DEST_DIR@
+prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${prefix}/@OBS_LIBRARY_DESTINATION@
-includedir=${prefix}/include
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
 
 Name: libobs
 Description: OBS Studio Library
-- 
2.22.0

